drivers/input/input_nunchuk.c

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2024 Bootlin
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nintendo_nunchuk
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/input/input.h>
+#include <zephyr/kernel.h>
+#include <zephyr/timing/timing.h>
+
+LOG_MODULE_REGISTER(input_nunchuk, CONFIG_INPUT_LOG_LEVEL);
+
+struct nunchuk_config {
+	struct i2c_dt_spec i2c_bus;
+	k_timeout_t polling_interval;
+};
+
+struct nunchuk_data {
+	const struct device *dev;
+	uint8_t joystick_x;
+	uint8_t joystick_y;
+	bool button_c;
+	bool button_z;
+	struct k_work_delayable work;
+};
+
+int nunchuk_read_registers(const struct device *dev, uint8_t *buffer)
+{
+	const struct nunchuk_config *cfg = dev->config;
+	int ret;
+	uint8_t value = 0;
+
+	ret = i2c_write_dt(&cfg->i2c_bus, &value, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	k_msleep(10);
+	ret = i2c_read_dt(&cfg->i2c_bus, buffer, 6);
```

Comment:
```suggestion
	ret = i2c_read_dt(&cfg->i2c_bus, buffer, sizeof(buffer));
```

Reply:
buffer is passed as an argument, so sizeof doesn't work here. I put a #define for it, I'm not totally sure about the name ?

Reply:
I see, how about passing an extra argument to `nunchuk_read_registers` with the buffer size? then you can do `nunchuk_read_registers(dev, buffer, sizeof(buffer));` and just use it as is in the function. Doesn't make any different just not thrilled with the function taking a pointer to a buffer and assuming its size.